test/testLottie.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <thorvg.h>
+#include <thorvg_lottie.h>
+#include <fstream>
+#include <cstring>
+#include "config.h"
+#include "catch.hpp"
+
+using namespace tvg;
+using namespace std;
+
+#ifdef THORVG_LOTTIE_LOADER_SUPPORT
+
+TEST_CASE("Lottie Slot", "[tvgLottie]")
+{
+    REQUIRE(Initializer::init(0) == Result::Success);
+
+    auto animation = LottieAnimation::gen();
+    REQUIRE(animation);
+
+    auto picture = animation->picture();
+    REQUIRE(picture->identifier == Picture::identifier);
+
+    const char* slotJson = R"({"gradient_fill":{"p":{"a":0,"k":[0,0.1,0.1,0.2,1,1,0.1,0.2,0.1,1]}}})";
+
+    //Slot override before loaded
+    REQUIRE(animation->override(slotJson) == Result::InsufficientCondition);
+
+    //Slot override
+    REQUIRE(picture->load(TEST_DIR"/lottieslot.json") == Result::Success);
+    REQUIRE(animation->override(slotJson) == Result::Success);
+
+    //Slot override with invalid JSON
+    REQUIRE(animation->override("") == Result::InsufficientCondition);
```

Comment:
Shouldn't it be InvalidArgument?

Reply:
<img width="655" alt="Screenshot 2024-03-06 at 3 57 43 PM" src="https://github.com/thorvg/thorvg/assets/11167117/6c627bfc-0ea2-4d7e-a1b2-962dcd79f4b0">

For now, returns InsufficientCondition in invalid JSON, should we change the result?

Reply:
Yes, we can change it if you think it makes sense.

Reply:
agree, #2033